src/scalarstats.jl

```diff
@@ -249,7 +249,7 @@ of the standard deviation under the assumption that the data is normally distrib
 """
 function mad(v::AbstractArray{T};
              center::Union{Real,Nothing}=nothing,
-             normalize::Union{Bool, Nothing}=nothing) where T<:Real
+             normalize::Union{Bool, Nothing}=nothing) where T<:Union{AbstractFloat,Integer}
```

Comment:
`AbstractIrrational` and `Rational` just error out. I can I try to see if we can manage `Rational`.

Reply:
It will error out with `Rational` due to
```
oftype(m, BigFloat("1.482602218505601860547076529360423431326703202590312896536266275245674447622701")
```
throwing `InexactError`... We can either put a catch and ask them to not use `normalize=true` with `T>:Rational` or use an approximation for `k`, and give a warning that they are losing precision in the process.

Should I verify the precision of `BigFloat` and give a warning if the precision exceeds the 256 hardcoded value?

Reply:
The `Rational` value for would be `6677046798800499//4503599627370496`. 
For `Irrational`, this works `mad!([ℯ, π])`, but `mad!([π])` or `mad!([ℯ])` won't as the single element `Vector` are `Array{Irrational{:π},1}` rather than `Vector{Float64}`. Could make it work by doing the `oftype` when `T <: Irrational` go to `Float64`.

Reply:
`Rational{T}` can use any precision depending on `T`, so there's no absolutely correct value. I'd just leave it alone (and `Irrational` too), but still use `Real`.